superset/security/manager.py

```diff
@@ -356,13 +356,12 @@ def rejected_tables(
         :param schema: The SQL database schema
         :returns: The rejected tables
         """
-
         query = sql_parse.ParsedQuery(sql)
 
         return {
             table
             for table in query.tables
-            if not self.can_access_datasource(database, table, schema)
+            if not self.can_access_datasource(database, table, table.schema or schema)
```

Comment:
I wonder if this line should be reverted, i.e., the `schema` is the fallback schema per the docstring and `can_access_datasource` should be changed to be, 

```
schema_perm = self.get_schema_perm(database, table.schema or schema)
```

Reply:
@john-bodley  good point, that is more appropriate place. updated the PR.